### ISSUE TITLE
Questions about training configurations

### ISSUE THREAD

**Issue by minghanz** [MehmetAygun/4D-PLS#11]
Hi, 

Thanks for open-sourcing the great work. I met some difficulty reproducing the result by training the model myself. The performance does not match that of the trained model you released. I am training on Tesla V100 so the GPU memory should not be a problem. My evaluation on the released trained model is close to the number reported in paper, so the evaluation code should be running properly. 

My first question is that some configurations used by your pretrained model (found in the `parameters.txt` file in the zip file of the pretrained weight) is not the same as the default as specified in `train_SemanticKitti.py`. The differences I found are listed below:
- `batch_num`: 8 (released trained weight) vs. 4 (train_SemanticKitti.py)
- `first_subsampling_dl`: 0.06 (released trained weight) vs. 0.12 (train_SemanticKitti.py)
- `in_features_dim`: 2 (released trained weight) vs. 3 (train_SemanticKitti.py)
- `n_frames`: 4 (released trained weight) vs. 2 (train_SemanticKitti.py)

Which setting shall I use? 

Second, shall I set `reinit_var` to `True` or `False` after pretraining for the 200 epochs? My understanding is that if you continue training from a pretrained model which is not trained with the variance head, then set `reinit_var` to `True` . If we resume training in the middle of the second stage training (in which case the variance head is also trained), then set `reinit_var` to `False`. Is it correct?

Third, do you pick the best model in a training session from the intermediate checkpoints, or you always use the latest model after the 1000 epochs of training? It seems that the intermediate checkpoints are not synchronized with the validation, thus we do not have information about the performance of the intermediate models?

Fourth, is there any way to know whether my model is being trained properly? Currently I have to train all the 1000 epochs and evaluate on the final model and compare with the released trained model, which is time-consuming. Do you have any suggestions on getting the rough idea in an earlier stage? (e.g., what is a proper loss or accuracy during training or validation at a smaller epoch number?)

Thanks for your help!

**Comment by MehmetAygun** [comment]
Hi, Thanks for your interest in our work. The settings inside of the parameters.txt should be the right ones. Regarding reinit_var yes your interpretation is right, if you didn't change anything for the architecture, you dont have to set reinit_var as True. 

For the 3rd and 4th questions, I was checking segmentation accuracy of validation for the pre-training stage, and for the full training checking the instance loss. In my experience, these lidar segmentation models really hard for overfitting as they do random sampling in every iteration. And you are right as it takes too much time I was not doing synchronized validation, I would suggest checking validation segmentation performance along with instance loss during training. But if I remember correctly 1000 epochs should be more than enough to get similar performance with the model that we shared.

Hope these details would help, otherwise please do not hesitate to ask more details.

**Comment by minghanz** [comment]
Thank you for the prompt reply. I will train following your instructions and come back to you if I have further questions!
